theseus/src/state/profiles.rs

```diff
@@ -674,10 +825,68 @@ impl Profiles {
         Ok(self)
     }
 
-    async fn read_profile_from_dir(path: &Path) -> crate::Result<Profile> {
+    async fn read_profile_from_dir(
+        path: &Path,
+        dirs: &DirectoryInfo,
+    ) -> crate::Result<Profile> {
         let json = fs::read(path.join(PROFILE_JSON_PATH)).await?;
         let mut profile = serde_json::from_slice::<Profile>(&json)?;
-        profile.path = PathBuf::from(path);
+
+        // Get name from stripped path
+        profile.path =
+            PathBuf::from(path.strip_prefix(dirs.profiles_dir().await)?);
+
         Ok(profile)
     }
+
+    pub fn sync_available_profiles_task(profile_path_id: ProfilePathId) {
+        tokio::task::spawn(async move {
+            let span = tracing::span!(
+                tracing::Level::INFO,
+                "sync_available_profiles_task"
+            );
+            let res = async {
+                let _span = span.enter();
+                let state = State::get().await?;
+                let dirs = &state.directories;
+                let mut profiles = state.profiles.write().await;
+
```

Comment:
what is this used for?

Reply:
Current debouncer watches *inside* profile folders and can update projects when the internal structure of the profile changes

I added the debouncer to watch the instances folder ( the profiles themselves) and this is triggered when that updates (eg: we are adding a whole new profile)